closes issue #19

grassesi opened the pull request.

clessig commented:
Hi @grassesi , many thanks for getting started at this. I just tried to run the changes and I get a long list of errors like this:

--- Logging error ---
Traceback (most recent call last):
  File "/usr/local/apps/python3/3.10.10-01/lib/python3.10/logging/__init__.py", line 1100, in emit
    msg = self.format(record)
  File "/usr/local/apps/python3/3.10.10-01/lib/python3.10/logging/__init__.py", line 943, in format
    return fmt.format(record)
  File "/lus/h2resw01/hpcperm/nacl/research/atmorep/develop/atmorep/atmorep/utils/logger.py", line 14, in format
    return super().format(record)
  File "/usr/local/apps/python3/3.10.10-01/lib/python3.10/logging/__init__.py", line 678, in format
    record.message = record.getMessage()
  File "/usr/local/apps/python3/3.10.10-01/lib/python3.10/logging/__init__.py", line 368, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/lus/h2resw01/hpcperm/nacl/research/atmorep/develop/atmorep/atmorep/core/train.py", line 237, in <module>
    train()
  File "/lus/h2resw01/hpcperm/nacl/research/atmorep/develop/atmorep/atmorep/core/train.py", line 227, in train
    cf.print()
  File "/lus/h2resw01/hpcperm/nacl/research/atmorep/develop/atmorep/atmorep/utils/utils.py", line 66, in print
    logger.info("{} : {}", key, value)
    
  Is there a specific python version of package that is required?

grassesi commented:
By default logging only supports old `%s` style formatting, but i used modern `"{}".format` style. To fix this I added support in `utils.logger.py` by using `logging.setLogRecordFactory` to set a custom `LogRecord` that supports both formatting styles. I added unit tests to test this behavior.
